clang/lib/Driver/Driver.cpp

```diff
@@ -2443,9 +2441,21 @@ void Driver::BuildInputs(const ToolChain &TC, DerivedArgList &Args,
           if (const char *Ext = strrchr(Value, '.'))
             Ty = TC.LookupTypeForExtension(Ext + 1);
 
+          // For SYCL, convert C-type sources to C++-type sources.
+          if (IsSYCL) {
```

Comment:
Should this be a switch instead? I would fear this getting unwieldy otherwise in pulldown/etc or if this got any more complicated.